spec/onceler_spec.rb

```diff
@@ -219,6 +219,46 @@ def my_method
     include_context "user cleanup"
   end
 
+  describe ".around(:once)" do
+    # don't use an ivar, they're not shared between arounds and examples
+    x = 0
+    around(:once) do |block|
+      x = 1
+      block.call
+      expect(x).to eq 2
+    end
+
+    before(:once) do
+      expect(x).to eq 1
+      x = 2
+    end
+
+    it "runs" do
+      expect(x).to eq 2
+    end
+  end
+
+  describe ".around(:once_and_each)" do
+    x = 0
+
+    around(:once_and_each) do |block|
+      x += 1
+      block.call
+    end
+
+    before(:once) do
+      expect(x).to eq 1
+    end
+
+    before(:each) do
+      expect(x).to eq 2
+    end
+
+    it "runs" do
```

Comment:
similarly could you have two examples to ensure that it increments :each time?

since run spec order isn't guaranteed, you could store prev_x before you increment, then your expectations all become `expect(x).to eq prev_x + 1`. and then perhaps at the very end:

```ruby
after(:all) do
  expect(x).to eq 3
end
```
to ensure it did all three (one :once and two :each)